tests/driver/runner_test.py

```diff
@@ -4489,6 +4489,76 @@ async def test_transform_stats_with_non_existing_path(self, es):
         es.transform.get_transform_stats.assert_called_once_with(transform_id=transform_id)
 
 
+class CreateIlmPolicyRunner(TestCase):
+
+    params = {
+        "policy-name": "my-ilm-policy",
+        "request-params": {"master_timeout": "30s", "timeout": "30s"},
+        "body": {
+            "policy": {
+                "phases": {"hot": {"min_age": "0ms", "actions": {"rollover": {"max_age": "30d"}, "set_priority": {"priority": 100}}}}
+            }
+        },
+    }
+
+    @mock.patch("elasticsearch.Elasticsearch")
+    @run_async
+    async def test_create_ilm_policy_with_request_params(self, es):
+        es.ilm.put_lifecycle.return_value = as_future({})
+        create_ilm_policy = runner.CreateIlmPolicy()
+        result = await create_ilm_policy(es, params=self.params)
+        self.assertEqual(1, result["weight"])
+        self.assertEqual("ops", result["unit"])
+        self.assertTrue(result["success"])
+
+        es.ilm.put_lifecycle.assert_called_once_with(
+            policy=self.params["policy-name"], body=self.params["body"], params=self.params["request-params"]
+        )
+
+    @mock.patch("elasticsearch.Elasticsearch")
+    @run_async
+    async def test_create_ilm_policy_without_request_params(self, es):
+        es.ilm.put_lifecycle.return_value = as_future({})
+        create_ilm_policy = runner.CreateIlmPolicy()
+        del self.params["request-params"]
```

Comment:
if we wanted to add any more tests to this class, this would be side-effecty i think. probably best to copy the params first before mangling